Not sure if this is a regression, but on this branch I cannot build `libiconv`:
```
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src/./iconv.c:713: undefined reference to `__errno_location'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src/./iconv.c:721: undefined reference to `__errno_location'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src/./iconv.c:693: undefined reference to `__errno_location'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src/./iconv.c:775: undefined reference to `__errno_location'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src/./iconv.c:783: undefined reference to `__errno_location'
../srclib/libicrt.a(progname.o): In function `set_program_name':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/progname.c:70: undefined reference to `program_invocation_short_name'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/progname.c:70: undefined reference to `program_invocation_short_name'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/progname.c:90: undefined reference to `program_invocation_name'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/progname.c:90: undefined reference to `program_invocation_name'
../srclib/libicrt.a(safe-read.o): In function `safe_read':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/safe-read.c:70: undefined reference to `__errno_location'
../srclib/libicrt.a(strerror.o): In function `rpl_strerror':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/srclib/strerror.c:60: undefined reference to `__errno_location'
../lib/.libs/libiconv.a(iconv.o): In function `libiconv_open':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/lib/./iconv.c:238: undefined reference to `__errno_location'
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/lib/./iconv.c:246: undefined reference to `__errno_location'
../lib/.libs/libiconv.a(iconv.o): In function `unicode_loop_convert':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/lib/./loop_unicode.h:(.text+0x81c): undefined reference to `__errno_location'
../lib/.libs/libiconv.a(iconv.o):/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/lib/./loop_unicode.h:347: more undefined references to `__errno_location' follow
../lib/.libs/libiconv.a(localcharset.o): In function `get_charset_aliases':
/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/lib/./../libcharset/lib/localcharset.c:217: undefined reference to `__isoc99_fscanf'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:78: iconv_no_i18n] Error 1
make[1]: Leaving directory '/tmp/nix-build-libiconv-1.15-aarch64-unknown-linux-android.drv-0/libiconv-1.15/src'
make: *** [Makefile:35: all] Error 2
builder for '/nix/store/86bvrw315q6s7hsp9fd3lgq4gzfzfs05-libiconv-1.15-aarch64-unknown-linux-android.drv' failed with exit code 2
cannot build derivation '/nix/store/74g33cc8px3galkxggx0w6a9wcd059jb-SDL-1.2.15-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/0nykzav51d0cb9lxqx2k7r81npa5g5lz-aarch64-unknown-linux-android-ghc-8.6.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/j66xgjdpv3ldka7n93ic3lpy8jpq28zj-SDL-0.6.7.0-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
error: build of '/nix/store/j66xgjdpv3ldka7n93ic3lpy8jpq28zj-SDL-0.6.7.0-aarch64-unknown-linux-android.drv' failed
```
(via `nix-build -A pkgsCross.aarch64-android-prebuilt.haskellPackages.SDL`)

--- a/pkgs/development/compilers/llvm/6/llvm.nix
+++ b/pkgs/development/compilers/llvm/6/llvm.nix
@@ -14,9 +14,6 @@
 , debugVersion ? false
 , enableManpages ? false
 , enableSharedLibraries ? true
-# Mesa requires AMDGPU target
-# BPF is used by bcc
-, enableTargets ? [ stdenv.hostPlatform stdenv.targetPlatform "AMDGPU" "BPF" ]
 }:
 
 let

--- a/pkgs/development/libraries/libpfm/default.nix
+++ b/pkgs/development/libraries/libpfm/default.nix
@@ -12,8 +12,8 @@ stdenv.mkDerivation rec {
   makeFlags = [
     "PREFIX=${placeholder "out"}"
     "LDCONFIG=true"
-    "ARCH=${stdenv.targetPlatform.uname.processor}"
-    "SYS=${stdenv.targetPlatform.uname.system}"
+    "ARCH=${stdenv.hostPlatform.uname.processor}"
+    "SYS=${stdenv.hostPlatform.uname.system}"
   ];
 
   NIX_CFLAGS_COMPILE = [ "-Wno-error" ];

--- a/pkgs/os-specific/linux/apparmor/default.nix
+++ b/pkgs/os-specific/linux/apparmor/default.nix
@@ -82,8 +82,8 @@ let
     prePatch = prePatchCommon + ''
       substituteInPlace ./libraries/libapparmor/swig/perl/Makefile.am --replace install_vendor install_site
       substituteInPlace ./libraries/libapparmor/swig/perl/Makefile.in --replace install_vendor install_site
-      substituteInPlace ./libraries/libapparmor/src/Makefile.am --replace "/usr/include/netinet/in.h" "${stdenv.cc.libc.dev}/include/netinet/in.h"
-      substituteInPlace ./libraries/libapparmor/src/Makefile.in --replace "/usr/include/netinet/in.h" "${stdenv.cc.libc.dev}/include/netinet/in.h"
+      substituteInPlace ./libraries/libapparmor/src/Makefile.am --replace "/usr/include/netinet/in.h" "${stdenv.lib.getDev stdenv.cc.libc}/include/netinet/in.h"
+      substituteInPlace ./libraries/libapparmor/src/Makefile.in --replace "/usr/include/netinet/in.h" "${stdenv.lib.getDev stdenv.cc.libc}/include/netinet/in.h"
     '';
     inherit patches;
 

